Sources/Vehicle/Templates/CPActionsTemplate.swift

```diff
@@ -0,0 +1,108 @@
+import CarPlay
+import Foundation
+import RealmSwift
+import Shared
+
+@available(iOS 15.0, *)
+final class CPActionsTemplate: CarPlayTemplateProvider {
+    private var listTemplate: CPListTemplate?
+    private var actionsToken: NotificationToken?
+    private let actions: Results<Action>
+
+    private var noActionsView: CPInformationTemplate = {
+        CPInformationTemplate(
+            title: L10n.About.Logo.title,
+            layout: .leading,
+            items: [
+                .init(title: L10n.CarPlay.NoActions.title, detail: nil),
+            ],
+            actions: []
+        )
+    }()
+
+    var template: CPTemplate
+
+    weak var interfaceController: CPInterfaceController?
+
+    init(actions: Results<Action>) {
+        self.actions = actions
+        self.template = CPTemplate()
+    }
+
+    func templateWillDisappear(template: CPTemplate) {
+        if template == self.template {
+            actionsToken?.invalidate()
+        }
+    }
+
+    func templateWillAppear(template: CPTemplate) {
+        if template == self.template {
+            update()
+        }
+    }
+
+    func update() {
+        template = list(for: actions)
```

Comment:
is there a reason to create a new template each time? i think it might make these objects a little easier to follow if you update in all cases rather than reset in some

Reply:
Not really, updated.